core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/SearchModelImpl.java

```diff
@@ -161,13 +162,12 @@ public int getAutocompleteTriggerThreshold() {
     return AUTOCOMPLETE_THRESHOLD;
   }
 
-  private List<String> getSearchTabs(Resource searchResource) {
+  private List<SearchTabDTO> getSearchTabs(Resource searchResource) {
```

Comment:
can we change this to SearchTabModel?

Reply:
SearchTabModel is for the searchTab component.
Here we need a DTO that is used to render searchtab related details (the url, the name (as an identifier utilized by FE)) in the search config json

Reply:
As discussed, if we adopt by the resource we should be fine, no?

Reply:
Actually we need the request object for creating the url (needed for pathTransformer::externalize).
However, found a solution for it. I am adapting to SearchTabModel resource with **modelFactory.getModelFromWrappedRequest(request, r, SearchTabModel.class)** instead. This creates and passes a ResourceOverridingRequestWrapper to the SearchTabModel ( which is then used in pathTransformer::externalize ).